packages/kbn-test/src/functional_test_runner/lib/mocha/decorate_mocha_ui.js

```diff
@@ -65,6 +66,10 @@ export function decorateMochaUi(lifecycle, context) {
             this._tags = [].concat(this._tags || [], tags);
           };
 
+          const relativeFilePath = this.file.replace(REPO_ROOT + sep, '');
```

Comment:
Rather than require awareness of `REPO_ROOT` in more places, and to keep things unambiguous + avoid using relative paths unnecessarily, we should use absolute paths here and resolve relative paths from CLIs based on the CWD at the time the CLI is invoked.

Reply:
I actually started working on it in my other PR like that, but switched it to relative because it makes everything a lot cleaner and simpler.

There are a bunch of places downstream that need to be relative. For example, when I output test metrics during CI (which will eventually be indexed in ES), the relative path is way better than an absolute path to a Jenkins workspace. When I match up metrics to test suites to figure out a test plan, I don't have to convert any paths because everything is just the relative path.

Doing it here (and for --include/exclude-file), means I only have to do it once at the beginning, and it also means things are consistent no matter where you run them, which is nice.

There are a few other things that end up a little nicer too, like this FTR output: ` info Only running suites (and their sub-suites) if they include the tag(s): [ 'test/functional/apps/discover/_doc_navigation.js' ]`

Reply:
It might be simpler in the places you're thinking of, but in general when dealing with paths throughout the repo it's a lot more complicated if that path may or may not be relative to any location. IMO, it's a lot simpler to manage paths in many places if we standardize on transforming paths into absolute paths for "internal" communication, and then convert them back to relative paths at display time.

> things are consistent no matter where you run them

I don't think that's a benefit when you're talking about a CLI. Making things consistent in this case means you can't use tab completion to pick paths which is how just about every other CLI works. That's the kind of consistency I think we should strive for.

Reply:
Oh, sorry, this part wasn't clear: `--include-file` and `--exclude-file` currently work no matter what the path you pass in looks like, as long as it's a valid path to the file based on your working directory, which is how a CLI would be expected to work.

```
➜  kibana git:(small-ci-test-suites) ✗ node scripts/functional_test_runner.js --include-file /Users/bseeders/code/kibana/test/functional/../functional/apps/dashboard/empty_dashboard.js --test-stats
 debg Loading config file from "/Users/bseeders/code/kibana/test/functional/config.js"
 debg Loading config file from "/Users/bseeders/code/kibana/test/common/config.js"
 info Config loaded
 info Only running suites (and their sub-suites) if they include the tag(s): [ 'test/functional/apps/dashboard/empty_dashboard.js' ]
{
  "testCount": 3,
  "excludedTests": []
}
```

Reply:
If you're in the `x-pack/plugins/foo` directory and pass `--include-file ../../tests/functional/apps/foo` it won't work right? That would normally work in most cases.

Reply:
Like this?

```
➜  apm git:(small-ci-test-suites) ✗ pwd
/Users/bseeders/code/kibana/x-pack/plugins/apm
➜  apm git:(small-ci-test-suites) ✗ node ../../../scripts/functional_test_runner.js --config ../../test/functional/config.js --include-file ../../test/functional/apps/apm/feature_controls/apm_spaces.ts --test-stats
 debg Loading config file from "/Users/bseeders/code/kibana/x-pack/test/functional/config.js"
 debg Loading config file from "/Users/bseeders/code/kibana/test/common/config.js"
 debg Loading config file from "/Users/bseeders/code/kibana/test/functional/config.js"
 info Config loaded
 info Only running suites (and their sub-suites) if they include the tag(s): [ 'x-pack/test/functional/apps/apm/feature_controls/apm_spaces.ts' ]
{
  "testCount": 4,
  "excludedTests": []
}
```

Reply:
Okay, I see, the relative paths are converted to absolute paths by `realpathSync()` and then back to paths that are relative from the repo root.

I can also live with the idea of absolute paths being converted to relative paths when they're converted to tags.

Can we at least convert the paths to absolute in a location where CWD is relatively safe to depend on (like the CLI) and not deep in the lib (setup_mocha.js) and then convert absolute paths to tags (relative paths from repo root) when we pass the paths to `filterSuiteByTags()`?

Also, can we please not use direct string manipulation and rely on the path methods like `path.resolve()` and `path.relative()`?

Reply:
Yep, that all sounds good to me. Will make those changes